### PR TITLE
feat: sorting and head menu in virtual scroll

### PR DESCRIPTION
**Pull request by haxxmaxx**
- using `HeadCellContent` inside `HeaderCell`
- moving `changeSortOrder` to context

### CODE REVIEW

File: src/table/virtualized-table/HeaderCell.tsx [hunk 1]
```diff
@@ -17,8 +18,10 @@ const HeaderCell = ({ index, style, data }: HeaderCellProps) => {
     columns,
     headerStyle: { ...applicableStyle },
   } = data;
+  const { layout } = useContextSelector(TableContext, (value) => value.baseProps);
   const datum = columns[index];
```
Comment (cbt1) [inline src/table/virtualized-table/HeaderCell.tsx:22]: Might as well rename `datum` to `column`

**Review by cbt1 — APPROVED** [review]
Cannot believe how easy that looked. Great work!

File: src/table/virtualized-table/HeaderCell.tsx (hunk 1 above)
Comment (haxxmaxx) [reply src/table/virtualized-table/HeaderCell.tsx:22]: agree

File: src/table/virtualized-table/HeaderCell.tsx [hunk 2]
```diff
@@ -37,7 +40,7 @@ const HeaderCell = ({ index, style, data }: HeaderCellProps) => {
         fontWeight: 'bold',
       }}
     >
-      <CellText singleLine>{datum.label}</CellText>
+      <HeadCellContent column={datum} columnIndex={index} isActive={isActive} areBasicFeaturesEnabled />
```
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/virtualized-table/HeaderCell.tsx:43]: yeah that would be nice, it is kind of used all over the place.

**Comment by haxxmaxx** [comment]
> Cannot believe how easy that looked. Great work!

Thanks! That was kind of all the prep work for it. Hopefully we can use the same way of implementing other data representations down the line, if we need them in both tables

**Pull request merged**